Database/compiler.lua

```diff
@@ -1122,53 +1131,135 @@ function QuestieDBCompiler:ValidateItems()
         --        end
         --    end
         --end
+        if count == TICKS_PER_YIELD_DEBUG then
+            count = 0
+            coroutine.yield()
+        end
+        count = count + 1
+    end
+    count = 0
+    for itemId, nonCompiledData in pairs(QuestieDB.itemData) do
+        local compiledData = validator.QueryValidator(itemId, QuestieDB.itemCompilerOrder)
+
+        for id, key in pairs(QuestieDB.itemCompilerOrder) do
+            local a = compiledData[id]
+            local b = nonCompiledData[QuestieDB.itemKeys[key]]
+
+            if type(a) == "number"  and abs(a-(b or 0)) > 0.2 then
+                Questie:Warning("Nonmatching number at " .. key .. "  " .. tostring(a) .. " ~= " .. tostring(b) .. " for ID: ".. itemId)
+                return
+            elseif type(a) == "string" and a ~= (b or "") then
+                Questie:Warning("Nonmatching string at " .. key .. "  " .. tostring(a) .. " ~= " .. tostring(b) .. " for ID: ".. itemId)
+                return
+            elseif type(a) == "table" then
+                if not equals(a, (b or {})) then
+                    Questie:Warning("Nonmatching table at " .. key .. "  " .. id  .. " for ID: ".. itemId)
+                    DevTools_Dump({
+                        ["Compiled Table:"] = a,
+                        ["Base Table:"] = b
+                    })
+                    return
+                end
+            end
+        end
+
+        if count == TICKS_PER_YIELD_DEBUG then
+            count = 0
+            coroutine.yield()
+        end
+        count = count + 1
     end
 
     validator.stream:finished()
     obj.stream:finished()
     npc.stream:finished()
-    Questie:Debug(Questie.DEBUG_INFO, "Finished items validation without issues!")
+    print(Questie.DEBUG_INFO, "Finished items validation without issues!")
 end
 
 function QuestieDBCompiler:ValidateQuests()
     local validator = QuestieDBCompiler:GetDBHandle(Questie.db.global.questBin, Questie.db.global.questPtrs, QuestieDBCompiler:BuildSkipMap(QuestieDB.questCompilerTypes, QuestieDB.questCompilerOrder))
 
-    for questId, _ in pairs(QuestieDB.questData) do
-        local toValidate = {validator.Query(questId, unpack(QuestieDB.questCompilerOrder))}
+    local playerLevel = UnitLevel("player")
+
+    local function getTbcLevel(questLevel, requiredLevel, playerLevel)
+        if (questLevel == -1) then
+            local level = playerLevel
+            if (requiredLevel > level) then
+                questLevel = requiredLevel;
+            else
+                questLevel = level;
+                -- We also set the requiredLevel to the player level so the quest is not hidden without "show low level quests"
+                requiredLevel = level;
+            end
+        end
+        return questLevel, requiredLevel
+    end
```

Comment:
This can also be created outside `QuestieDBCompiler:ValidateQuests`

Reply:
It is only used in here so id rather not have it outside

Reply:
furthermore a much nicer way of fixing this is in the refactor/DB branch, so i made a ticket for it and we'll fix it in there.